### PR TITLE
check if rust/cargo is installed before installting better-tree

### PR DESCRIPTION
**Pull request by harshRn**
check if rust/cargo is installed before installting better-tree

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,7 +48,14 @@ RESET			:= $(shell echo "\033[0m")
 
 all: install
 
-$(TARGET):
+# Check if cargo is installed
+pre:
+ifndef RUST_CRGO
```
Comment (tesla33io) [inline Makefile]: typo, it should be `RUST_CARGO`

**Comment by tesla33io** [comment]
I appreciate your effort, but unfortunately, your changes did not pass my tests. Relying solely on checking an environment variable to determine if a program is installed may not be the most reliable method. I suggest exploring the use of `which` or `command` utilities for a more accurate check. These tools can provide insight into how to address this issue effectively.

For context, here's the output of my test after correcting a typo:

```bash
$> make
Makefile:54: *** "Cargo is not installed. Please install the Rust toolchain (https://www.rust-lang.org/tools/install)".  Stop.
$> cargo
Rust's package manager

Usage: cargo [+toolchain] [OPTIONS] [COMMAND]
       cargo [+toolchain] [OPTIONS] -Zscript <MANIFEST_RS> [ARGS]...
... more output ...
```

This output indicates that while Cargo is indeed installed, the Makefile's method for detecting it is not functioning as intended. Utilizing `which` or `command` could potentially solve this by verifying the presence of Cargo more reliably.

File: Makefile (hunk 1 above)
Comment (harshRn) [reply Makefile]: I hope you see that this was pre-existing and a less than trivial issue.

**Comment by harshRn** [comment]
also, `RUST_CRGO`(or `RUST_CARGO` if we do end up re-naming it) is not an environment var we are looking for in the system, It's something that we init earlier in the Makefile. and you would even see that it is indeed using the `command` utility as you have requested. 
FWIW, I changed your variable's name to `RUST_CARGO`.  I believe this is in good shape to merge. Cheers!

**Comment by tesla33io** [comment]
Indeed, I messed up with my test. It's actually working. Sorry for my mistake.

**Pull request merged**